powerlibs/aws/process_status_notifier/__init__.py

```diff
@@ -33,7 +33,9 @@ def __enter__(self, *args):
 
     def __exit__(self, exc_type, exc_value, traceback):
         if exc_type:
-            self.notify('failed', payload={**self.payload, 'message': str(exc_value)})
+            payload = self.payload
```

Comment:
Change this to:

```python
payload = dict(self.payload)
```

This way the `self.payload` won't be tainted by this method.
(I know the effects are minimum, since it's the `__exit__` method, but...)